### PR TITLE
8333754: Add a Test against ECDSA and ECDH NIST Test vector

### PR DESCRIPTION
**Pull request by sisahoo**
Tests added against ECDSA and ECDH NIST Test vector.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333754](https://bugs.openjdk.org/browse/JDK-8333754): Add a Test against ECDSA and ECDH NIST Test vector (**Bug** - P3)

### Reviewers
 * [Anthony Scarpino](https://openjdk.org/census#ascarpino) (@ascarpino - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19715/head:pull/19715` \
`$ git checkout pull/19715`

Update a local copy of the PR: \
`$ git checkout pull/19715` \
`$ git pull https://git.openjdk.org/jdk.git pull/19715/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19715`

View PR using the GUI difftool: \
`$ git pr show -t 19715`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19715.diff">https://git.openjdk.org/jdk/pull/19715.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19715#issuecomment-2167313949)

### CODE REVIEW

File: test/jdk/sun/security/ec/ECDHPrimitive.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import javax.crypto.*;
+
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8189189
+ * @summary Test ECDH primitive operations
+ * @library /test/lib
+ * @run main ECDHPrimitive
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDHPrimitive.java:40]: disableNative removed from @run tag

File: test/jdk/sun/security/ec/ECDHPrimitive.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import java.io.*;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.*;
+import java.security.spec.*;
+import java.util.*;
+import javax.crypto.*;
+
+import jdk.test.lib.Asserts;
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDHPrimitive.java:33]: Usage of Asserts and removed corresponding local code

File: test/jdk/sun/security/ec/ECDHPrimitive.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+
+    private static final Map<String, String> NAME_MAP = Map.of(
+            "P-256", "secp256r1",
+            "P-384", "secp384r1",
+            "P-521", "secp521r1"
+    );
+
+    public static void main(String[] args) throws Exception {
+        Path testFile = Path.of(System.getProperty("test.src"), "KAS_ECC_CDH_PrimitiveTest.txt");
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDHPrimitive.java:52]: Referring Test vector from SRC path

File: test/jdk/sun/security/ec/ECDSAPrimitive.java [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8189189 8147502 8295010
+ * @summary Test ECDSA primitive operations
+ * @library /test/lib
+ * @modules java.base/sun.security.ec java.base/sun.security.ec.point
+ *          java.base/sun.security.util java.base/sun.security.util.math
+ * @run main ECDSAPrimitive
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDSAPrimitive.java:44]: disableNative removed from @run tag

File: test/jdk/sun/security/ec/ECDSAPrimitive.java [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+import java.nio.file.Path;
+import java.security.*;
+import java.security.spec.*;
+import java.util.*;
+
+import sun.security.ec.*;
+import sun.security.ec.point.*;
+import sun.security.util.ArrayUtil;
+import sun.security.util.math.*;
+import jdk.test.lib.Asserts;
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDSAPrimitive.java:35]: Use of Asserts and removed corresponding local code.

File: test/jdk/sun/security/ec/ECDSAPrimitive.java [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+    );
+    private static final Set<String> DIGEST_NAME_SET = Set.of(
+            "SHA-224",
+            "SHA-256",
+            "SHA-384",
+            "SHA-512"
+    );
+
+    public static void main(String[] args) throws Exception {
+        Path siggenFile = Path.of(System.getProperty("test.src"), "SigGen-1.txt");
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDSAPrimitive.java:61]: Referring Test vector from SRC path.

File: test/jdk/sun/security/ec/ECDSAPrimitive.java [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+        IntegerModuloP u1 = e.multiply(sInverse);
+        IntegerModuloP u2 = r.multiply(sInverse);
+
+        byte[] u1Bytes = u1.asByteArray(length);
+        byte[] u2Bytes = u2.asByteArray(length);
+        AffinePoint publicKeyPoint = ECDSAOperations.toAffinePoint(publicKey,
+                ecOps.getField());
+        MutablePoint R = ecOps.multiply(publicKeyPoint, u2Bytes);
+        AffinePoint a1 = ops.basePointMultiply(u1Bytes);
+        MutablePoint p2 = new ProjectivePoint.Mutable(
```
Comment (sisahoo) [inline test/jdk/sun/security/ec/ECDSAPrimitive.java:255]: Compile time issue fixed here.

**Review by ascarpino — APPROVED** [review]
The changes look good to me.

**Pull request closed without merging**